src/test/ui/rustdoc/doc-test-attr.rs

```diff
@@ -0,0 +1,16 @@
+#![crate_type = "lib"]
+#![deny(invalid_doc_attributes)]
+
+#![doc(test)]
+//~^ ERROR `#[doc(test(...)]` takes a list of attributes
+//~^^ WARN this was previously accepted by the compiler
+#![doc(test = "hello")]
+//~^ ERROR `#[doc(test(...)]` takes a list of attributes
+//~^^ WARN this was previously accepted by the compiler
+#![doc(test(a))]
+//~^ ERROR unknown `doc(test)` attribute `a`
+//~^^ WARN this was previously accepted by the compiler
+#![doc(test(no_crate_inject))]
+#![doc(test(attr(deny(warnings))))]
```

Comment:
I'm wondering if it's worth having a separate check-pass test to ensure these do definitely work. Could you add one?

Reply:
I'll add one then.